android/library/src/main/java/io/logto/android/authflow/webview/WebViewAuthFlow.kt

```diff
@@ -43,4 +71,36 @@ class WebViewAuthFlow {
         )
         return UrlUtil.appendQueryParameters(baseUrl.buildUpon(), parameters).toString()
     }
+
+    private fun authenticate(
+        authorizationCode: String,
+        authenticationCallback: AuthenticationCallback
+    ) {
+        val call = logtoClient.getCredential(
+            logtoConfig.redirectUri,
+            authorizationCode,
+            AuthConstant.GrantType.AUTHORIZATION_CODE,
+            logtoConfig.clientId,
+            codeVerifier,
+        )
+
+        call.enqueue(object : Callback<Credential> {
+            override fun onResponse(call: Call<Credential>, response: Response<Credential>) {
+                val credential = response.body()
+                if (credential != null) {
+                    authenticationCallback.onSuccess(credential)
+                    return
+                }
+                authenticationCallback.onFailed(Error("fetch credential error"))
+            }
+
+            override fun onFailure(call: Call<Credential>, t: Throwable) {
+                authenticationCallback.onFailed(Error("authentication request error"))
+            }
+        })
```

Comment:
may be [combine](https://loopcupcakes.com/blog/network-retrofit-coroutines) retrofit with [coroutines](https://kotlinlang.org/docs/coroutines-overview.html)?

Reply:
TBD: [LOG-84](https://linear.app/silverhand/issue/LOG-84/use-coroutines-in-project)